src/annotator/anchoring/test/types-test.js

```diff
@@ -19,46 +21,335 @@ describe('types', function () {
+      });
+    });
+
+    describe('#toSelector', () => {
+      beforeEach(() => {
+        fakeSerialize.returns({
+          start: '/div[1]',
+          startOffset: 0,
+          end: '/div[1]',
+          endOffset: 1,
+        });
+      });
+
+      function rangeSelectorResult() {
+        return {
+          type: 'RangeSelector',
+          startContainer: '/div[1]',
+          startOffset: 0,
+          endContainer: '/div[1]',
+          endOffset: 1,
+        };
+      }
+
+      it('returns a RangeSelector', () => {
+        const range = new RangeAnchor(container, new Range());
+        assert.deepEqual(range.toSelector({}), rangeSelectorResult());
+      });
+
+      it('returns a RangeSelector if options are missing', () => {
+        const range = new RangeAnchor(container, new Range());
+        assert.deepEqual(range.toSelector(), rangeSelectorResult());
+      });
+    });
+  });
+
+  describe('TextPositionAnchor', () => {
+    let fakePosFromRange;
+    let fakePosToRange;
+
+    beforeEach(() => {
+      fakePosFromRange = sinon.stub();
+      fakePosToRange = sinon.stub();
+      $imports.$mock({
+        'dom-anchor-text-position': {
+          fromRange: fakePosFromRange,
+          toRange: fakePosToRange,
+        },
+      });
+    });
+
+    afterEach(() => {
+      $imports.$restore();
+    });
+
+    function createTextPositionAnchor() {
+      return new TextPositionAnchor(container, 0, 0);
+    }
+
+    describe('#fromRange', () => {
+      it('returns a TextPositionAnchor instance', () => {
+        fakePosFromRange.returns({
+          start: 0,
+          end: 1,
+        });
+        const anchor = TextPositionAnchor.fromRange(container, new Range());
+        assert.calledOnce(fakePosFromRange);
+        assert.instanceOf(anchor, TextPositionAnchor);
+      });
+    });
+
+    describe('#fromSelector', () => {
+      it('returns a TextPositionAnchor instance', () => {
+        const anchor = TextPositionAnchor.fromSelector(container, {
+          start: 0,
+          end: 0,
+        });
+        assert.instanceOf(anchor, TextPositionAnchor);
+      });
+    });
+
+    describe('#toSelector', () => {
+      it('returns a TextPositionSelector', () => {
+        const anchor = createTextPositionAnchor();
+        assert.deepEqual(anchor.toSelector(), {
+          type: 'TextPositionSelector',
+          start: 0,
+          end: 0,
+        });
+      });
+    });
+
+    describe('#toRange', () => {
+      it('returns a range object', () => {
+        const anchor = createTextPositionAnchor();
+        fakePosToRange.returns('fake range');
+        assert.equal(anchor.toRange(), 'fake range');
+        assert.calledWith(fakePosToRange, container, { start: 0, end: 0 });
+      });
+    });
+
+    describe('integration tests', () => {
+      beforeEach(() => {
+        // restore dom-anchor-text-position to test third party lib integration
+        $imports.$restore();
```

Comment:
This will restore _all_ mocks, not just `dom-anchor-text-position`. Use `$imports.$restore({ 'dom-anchor-text-position': true })` for that.